src/main/java/org/spongepowered/api/event/entity/ai/AITargetEvent.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.entity.ai;
+
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.entity.living.Agent;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.entity.living.TargetAgentEvent;
+
+/**
+ * Fires before the target of an {@link Agent} is changed.
+ */
+public interface AITargetEvent extends TargetAgentEvent, Cancellable
+{
+
+    /**
+     * Gets the {@link Agent}s new target {@link Entity}.
+     * @return The target
+     */
+    Entity getAITarget();
+
+    /**
+     * Sets the {@link Agent}s new target {@link Entity}
+     *
+     * @param target The target
+     */
+    void setAITarget(Entity target);
```

Comment:
What's the difference between this and https://github.com/SpongePowered/SpongeAPI/blob/14f459d824e3b0216a2ec0c4683c2c06fc80404c/src/main/java/org/spongepowered/api/entity/living/Agent.java#L59

Reply:
The target (when not cancelled) will be set after the event.